code/game/machinery/autolathe.dm

```diff
@@ -640,8 +640,8 @@
 
 
 	//And if there's any remainder, we eject that as a shard
-	if (remainder)
```

Comment:
this was valid before, shards contain a variable quantity of the material

besides this is going to be a non integer value in most cases, you can't iterate through it